code_search/code-search.ipynb

```diff
@@ -0,0 +1,451 @@
+    "echo \"Google Cloud SDK Info: \" && gcloud --version && echo\n",
+    "echo \"Kubectl Version Info: \" && kubectl version && echo\n",
+    "echo \"Ksonnet Version Info: \" && ks version"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Get the Source Code\n",
+    "\n",
+    "Let us clone the source code for Code Search from [kubeflow/examples](https://github.com/kubeflow/examples)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%bash\n",
+    "\n",
+    "# NOTE: This cell must only be run once or the clone will complain.\n",
+    "git clone --depth=1 https://github.com/kubeflow/examples"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Install dependencies\n",
+    "\n",
+    "Let us install all the Python dependencies. Note that everything must be done with `Python 2`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%bash\n",
+    "\n",
+    "pip2 install -r examples/code_search/src/requirements.txt\n",
+    "pip2 install https://github.com/activatedgeek/batch-predict/tarball/fix-value-provider"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Configure Variables\n",
+    "\n",
+    "This involves setting up the Ksonnet application as well as utility environment variables for various CLI steps."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Configuration Variables. Modify as desired.\n",
+    "\n",
+    "PROJECT = 'kubeflow-dev'\n",
+    "TARGET_DATASET = 'code_search'\n",
+    "WORKING_DIR = 'gs://kubeflow-examples/t2t-code-search/20180807'\n",
+    "\n",
+    "WORKER_MACHINE_TYPE = 'n1-highcpu-32'\n",
+    "NUM_WORKERS = 16\n",
+    "\n",
+    "# DO NOT MODIFY. The following statements are needed to support\n",
+    "# environment variables in a bash cell.\n",
+    "%env PROJECT $PROJECT\n",
+    "%env TARGET_DATASET $TARGET_DATASET\n",
+    "%env WORKING_DIR $WORKING_DIR\n",
+    "%env WORKER_MACHINE_TYPE $WORKER_MACHINE_TYPE\n",
+    "%env NUM_WORKERS $NUM_WORKERS"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Add the Kubernetes environment to the Ksonnet application and update parameters."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%bash\n",
+    "\n",
+    "ks env add code-search --context=$(kubectl config current-context) --namespace=kubeflow\n",
+    "\n",
+    "# TODO: refactor Kubeflow application to allow setting common variables.\n",
+    "# ks param set t2t-job workingDir ${WORKING_DIR}"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Pre-Processing Github Files\n",
```

Comment:
Lets add a cell before this to query BigQuery to show the first 10 rows. We'd like to allow people to inspect the data.